pkg/controller/common/reconciler.go

```diff
@@ -30,6 +33,36 @@ func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci network
 
 	exposed := ci.GetSpec().Visibility == networkingv1alpha1.IngressVisibilityExternalIP
 	if exposed {
+		// get the namespace where operator has been installed
+		ns, err := k8sutil.GetOperatorNamespace()
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
```

Comment:
Can we use the usual `IsNotFound` error check from K8s API packages?

Reply:
return type of `k8sutil.GetOperatorNamespace()` is 
`func GetOperatorNamespace() (string, error) {`
so we cannot use `IsNotFound` of K8s API packages

Reply:
Why not? The error is just an apierror isn't it?

Reply:
Returned error is builtin error not from `k8s.io/apimachinery/pkg/api/errors`

Reply:
Hm true, this is from a file operation.

I'm wondering if the operator is even installed into knative-serving when installed via OLM. Can you doublecheck that please? We might need to hardcode `knative-serving` here for now as this operator is now in theory not tied to a specific knative-serving installation. As such, I'd drop this detection logic for now.

Reply:
We can however use a similar logic to that in `MakeRoutes` to figure out the correct namespace here!

Reply:
Yes when installed via OLM everything works perfect
But if at all run using `operator-sdk up local --namespace=""` than `k8sutil.GetOperatorNamespace()` returns empty so in that case we cannot proceed further so to handle that particularly added if condition

Reply:
As per the discussion as of now hardcoded namespace to `knative-serving-ingress`